deepinterpolation/generator_collection.py

```diff
@@ -536,126 +398,163 @@ def __data_generation__(self, local_x, local_y, local_z, local_t):
         else:
             self.pre_frame = self.json_data["pre_frame"]
             self.post_frame = self.json_data["post_frame"]
 
-        self.pre_post_omission = self.json_data["pre_post_omission"]
-        self.start_frame = self.json_data["start_frame"]
-        self.steps_per_epoch = self.json_data["steps_per_epoch"]
-
-        # This is used to limit the total number of samples
-        # -1 means to take all and is the default fall back
-
         if "total_samples" in self.json_data.keys():
             self.total_samples = self.json_data["total_samples"]
         else:
             self.total_samples = -1
 
-        # This is compatible with negative frames
+        if "randomize" in self.json_data.keys():
+            self.randomize = self.json_data["randomize"]
+        else:
+            self.randomize = 1
+
+        if "pre_post_omission" in self.json_data.keys():
+            self.pre_post_omission = self.json_data["pre_post_omission"]
+        else:
+            self.pre_post_omission = 0
+
+        # load parameters that are related to training jobs
+        self.batch_size = self.json_data["batch_size"]
+        self.steps_per_epoch = self.json_data["steps_per_epoch"]
+
+        # Loading limit parameters
+        self.start_frame = self.json_data["start_frame"]
         self.end_frame = self.json_data["end_frame"]
 
-        self.nb_probes = 384
+        # start_frame starts at 0
+        # end_frame is compatible with negative frames. -1 is the last
+        # frame.
 
-        self.raw_data = np.memmap(self.raw_data_file, dtype="int16")
+        # We initialize the epoch counter
+        self.epoch_index = 0
+
+    def update_end_frame(self, total_frame_per_movie):
+        """Update end_frame based on the total number of frames available.
+        This allows for truncating the end of the movie when end_frame is
+        negative."""
 
+        # This is to handle selecting the end of the movie
         if self.end_frame < 0:
-            self.img_per_movie = (
-                int(self.raw_data.size / self.nb_probes)
-                + 1
-                + self.end_frame
-                - self.start_frame
-                - self.post_frame
-                - self.pre_post_omission
-            )
-        elif int(self.raw_data.size / self.nb_probes) < self.end_frame:
-            self.img_per_movie = (
-                int(self.raw_data.size / self.nb_probes)
-                - self.start_frame
-                - self.post_frame
-                - self.pre_post_omission
-            )
-        else:
-            self.img_per_movie = self.end_frame + 1 - self.start_frame
+            self.end_frame = total_frame_per_movie+self.end_frame
+        elif total_frame_per_movie <= self.end_frame:
+            self.end_frame = total_frame_per_movie-1
 
-        self.total_frame_per_movie = int(self.raw_data.size / self.nb_probes)
+    def calculate_list_samples(self, total_frame_per_movie):
```

Comment:
Looks like this method might be private. If so, it should begin with _

Reply:
True updated